### PR TITLE
EUSS Cycle 3 Electric Cooking Equipment

### PR DESCRIPTION
**Pull request by mpraprost**
Pull request overview
---------------------

EUSS Cycle 3 Measure - Electric Cooking Equipment 
Measure documentation updated with 10k results
NOTE - still working on wrapping up unit tests, but wanted to get this PR in for review

### Pull Request Author: Marley

This pull request makes changes to (select all the apply):
 - [ ] Documentation
 - [ ] Infrastructure (includes singularity image, buildstock batch, dependencies, continuous integration tests)
 - [ ] Sampling
 - [ ] Workflow Measures
 - [X] Upgrade Measures
 - [ ] Reporting Measures
 - [ ] Postprocessing

Author pull request checklist:

 - [X] Tagged the pull request with the appropriate label (documentation, infrastructure, sampling, workflow measure, upgrade measure, reporting measure, postprocessing) to help categorize changes in the release notes.
 - [ ] Added tests for new measures
 - [X] Updated measure .xml(s)
 - [ ] Register values added to `comstock_column_definitions.csv`
 - [X] Both `options_lookup.tsv` files updated
 - [X] 10k+ test run
 - [ ] Change documentation written
 - [X] Measure documentation written
 - [ ] ComStock documentation updated
 - [ ] Changes reflected in example `.yml` files
 - [ ] Changes reflected in `README.md` files
 - [ ] Added 'See ComStock License' language to first two lines of each code file
 - [ ] All new and existing tests pass the CI

### Review Checklist

This will not be exhaustively relevant to every PR.
 - [ ] Perform a code review on GitHub
 - [ ] All related changes have been implemented: data and method additions, changes, tests
 - [ ] If fixing a defect, verify by running develop branch and reproducing defect, then running PR and reproducing fix
 - [ ] Reviewed change documentation
 - [ ] Ensured code files contain License reference
 - [ ] Results differences are reasonable
 - [ ] CI status: all tests pass

#### ComStock Licensing Language - Add to Beginning of Each Code File
```
# ComStock™, Copyright (c) 2023 Alliance for Sustainable Energy, LLC. All rights reserved.
# See top level LICENSE.txt file for license terms.
```

### CODE REVIEW

File: resources/measures/set_primary_kitchen_equipment/measure.rb [hunk 1]
```diff
@@ -188,13 +188,28 @@ def run(model, runner, user_arguments)
...
     # make hash of electric equipment
     appliance_electric_power_hash = Hash[li_appliance_types.zip li_appliance_electric_power_kW]
     # make list of gas equipment power (in kW)
-    li_appliance_gas_power_kW = [25.800,20.500,48.100,16.400,52.500,61.500]
+    li_appliance_gas_power_kW = [28.136,26.377,23.447,12.896,42.497,58.617]
     # make hash of gas equipment
     appliance_gas_power_hash = Hash[li_appliance_types.zip li_appliance_gas_power_kW]
+   
+    # fraction latent always set to 0.1
+    frac_latent = 0.1
```
Comment (ChristopherCaradonna) [inline resources/measures/set_primary_kitchen_equipment/measure.rb:200]: is this change reflected in the baseline as well?

File: resources/measures/set_primary_kitchen_equipment/measure.rb [hunk 2]
```diff
@@ -188,13 +188,28 @@ def run(model, runner, user_arguments)
     # make hash of equipment quantities
     appliance_quantity_hash = Hash[li_appliance_types.zip li_appliance_quantities]
     # make list of electric equipment power (in kW)
-    li_appliance_electric_power_kW = [11.000,12.000,22.000,15.000,12.000,15.000]
+    li_appliance_electric_power_kW = [10.815,17.116,14.009,12.104,21.014,26.964]
```
Comment (ChristopherCaradonna) [inline resources/measures/set_primary_kitchen_equipment/measure.rb]: Weren't these changes all reflected in a separate PR: https://github.com/NREL/ComStock/pull/66

File: resources/measures/upgrade_ppl_electric_kitchen_equipment/measure.rb [hunk 3]
```diff
@@ -0,0 +1,129 @@
...
+      {:equip_type=>"griddle", :elec_design_level_w=>17116.0, :frac_latent=>0.1, :frac_radiant=>0.39, :frac_lost=>0.41},
+      {:equip_type=>"oven", :elec_design_level_w=>12104.0, :frac_latent=>0.1, :frac_radiant=>0.22, :frac_lost=>0.58},
+      {:equip_type=>"range", :elec_design_level_w=>21014.0, :frac_latent=>0.1, :frac_radiant=>0.1, :frac_lost=>0.8},
+      {:equip_type=>"steamer", :elec_design_level_w=>26964.0, :frac_latent=>0.1, :frac_radiant=>0.1, :frac_lost=>0.79}
+    ]
+
+    # measure not applicable if building does not have a kitchen
+    if num_kitchens == 0
+      runner.registerAsNotApplicable("Building does not have a kitchen; measure is not applicable.")
+    end
```
Comment (ChristopherCaradonna) [inline resources/measures/upgrade_ppl_electric_kitchen_equipment/measure.rb]: add return true

File: resources/measures/upgrade_ppl_electric_kitchen_equipment/tests/kitchen_equip_measure_test.rb [hunk 4]
```diff
@@ -0,0 +1,479 @@
...
+
+    # Make an empty model
+    model = OpenStudio::Model::Model.new
+
+    # Get arguments and test that they are what we are expecting
+    arguments = measure.arguments(model)
+    assert_equal(0, arguments.size)
+  end
+
+  def test_r_value_cz_3a_single_pane
```
Comment (ChristopherCaradonna) [inline resources/measures/upgrade_ppl_electric_kitchen_equipment/tests/kitchen_equip_measure_test.rb]: need to update measure tests.

**Review by ChristopherCaradonna — CHANGES_REQUESTED** [review]
Looks good. I added a few comments in the measure doc, and some minor comments in the code.

File: resources/measures/set_primary_kitchen_equipment/measure.rb (hunk 2 above)
Comment (mpraprost) [reply resources/measures/set_primary_kitchen_equipment/measure.rb]: gone after updating from main

File: resources/measures/upgrade_ppl_electric_kitchen_equipment/measure.rb (hunk 3 above)
Comment (mpraprost) [reply resources/measures/upgrade_ppl_electric_kitchen_equipment/measure.rb]: done

File: resources/measures/upgrade_ppl_electric_kitchen_equipment/tests/kitchen_equip_measure_test.rb (hunk 4 above)
Comment (mpraprost) [reply resources/measures/upgrade_ppl_electric_kitchen_equipment/tests/kitchen_equip_measure_test.rb]: in progress

**Pull request closed without merging**